main.go

```diff
@@ -104,6 +104,17 @@ func main() {
 		setupLog.Error(err, "unable to create controller", "controller", "HostDeviceNetwork")
 		os.Exit(1)
 	}
+
+	upgradeLogger := ctrl.Log.WithName("controllers").WithName("Upgrade")
+
+	if err = (&controllers.UpgradeReconciler{
+		Client: mgr.GetClient(),
+		Log:    upgradeLogger,
```

Comment:
nit: can you directly create the log here ? for consistency with other controllers

Reply:
The idea was to create a single logger for all upgrade-related entities for easy logs filtering. That way the logger is created in main and propagated to all other components